Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+                    ExportActivityAsTestStep(test, (Activity)actIdent.IdentifiedActivity);
+
+                return true;
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export the Activities Group to QC/ALM", ex);
+                return false;
+            }
+        }
+
+        private static bool ExportActivityAsTestStep(Test test, Activity activity)
+        {
+            DesignStepFactory stepF;
+            DesignStep step = null;
+            List stepsList;
+
+            stepF = test.DesignStepFactory;
+
+            if (string.IsNullOrEmpty(activity.ExternalID) == false)
+            {
+                //look for existing step                    
+                stepsList = stepF.NewList("");
+                if (stepsList != null && stepsList.Count > 0)
+                    foreach (DesignStep s in stepsList)
+                        if (s.ID.ToString() == activity.ExternalID)
+                        {
+                            step = s;//step already exist 
+                            break;
+                        }
+            }
+
+            if (step == null)
+            {
+                //create new step
+                step = stepF.AddItem(System.DBNull.Value);
+            }
+
+            step.StepName = activity.ActivityName;
+            string descriptionTemplate =
+                "<html><body><div align=\"left\"><font face=\"Arial\"><span style=\"font-size:8pt\"><<&Description&&>><br /><<&Parameters&>><br /><<&Actions&>></span></font></div></body></html>";
+            string description = descriptionTemplate.Replace("<<&Description&&>>",activity.Description);
+            StepParams testParams = test.Params;
+            string paramsSigns = string.Empty;
+            if (activity.Variables.Count > 0)
+            {
+                paramsSigns = "<br />Parameters:<br />";
+                foreach (VariableBase var in activity.Variables)
+                {
+                    paramsSigns += "&lt;&lt;&lt;" + var.Name.ToLower() + "&gt;&gt;&gt;<br />";
+                    //try to add the parameter to the test case parameters list
+                    try
+                    {
+                        testParams.AddParam(var.Name.ToLower(), "String");
+                        testParams.Save();
+                        test.Post();
+                    }
+                    catch (Exception ex) { Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex); }
+                }
+            }
+            description = description.Replace("<<&Parameters&>>", paramsSigns);
+
+            string actsDesc = string.Empty;
+            if (activity.Acts.Count > 0)
+            {
+                actsDesc = "Actions:<br />";
+                foreach (Act act in activity.Acts)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'foreach' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)